code/parachain/frame/pablo/src/test/dual_asset_constant_product_tests_new.rs

```diff
@@ -301,18 +380,137 @@ mod do_swap {
 			);
 		});
 	}
+
+	proptest! {
+		#![proptest_config(ProptestConfig::with_cases(FIRST_ASSET_WEIGHTS.len() as u32))]
+
+		/// Ensure the outcome of the `do_swap` function respects token pool weights other than 50/50
+		#[test]
+		fn should_respect_non_50_50_weight(first_asset_weight in first_asset_weight()) {
+			new_test_ext().execute_with(|| {
+				process_and_progress_blocks::<Pablo, Test>(1);
+
+				// BTC/USDT Pool with a 0% fee
+				let pool_id =
+					create_pool_from_config(PoolInitConfiguration::DualAssetConstantProduct {
+						owner: ALICE,
+						assets_weights: dual_asset_pool_weights(BTC, first_asset_weight, USDT),
+						fee: Permill::zero(),
+					});
+
+				// Mint tokens for adding liquidity
+				let initial_btc = BTC::units(512_000);
+				let initial_usdt = USDT::units(512_000);
+				assert_ok!(Tokens::mint_into(BTC, &ALICE, initial_btc));
+				assert_ok!(Tokens::mint_into(USDT, &ALICE, initial_usdt));
+
+				// Add liquidity
+				assert_ok!(Pablo::add_liquidity(
+					Origin::signed(ALICE),
+					pool_id,
+					BTreeMap::from([(BTC, initial_btc), (USDT, initial_usdt)]),
+					0,
+					false
+				));
+
+				// Mint tokens for swap
+				let usdt_to_swap = AssetAmount::new(USDT, USDT::units(256_000));
+				assert_ok!(Tokens::mint_into(USDT, &BOB, usdt_to_swap.amount));
+				assert_eq!(Tokens::balance(USDT, &BOB), usdt_to_swap.amount);
+
+				// Do swap
+				assert_ok!(Pablo::do_swap(&BOB, pool_id, usdt_to_swap, ZERO_BTC, false));
+
+				let expected_btc_post_buy =	compute_out_given_in(
```

Comment:
I think we may still need some hard coded values for each weight so that we detect any regressions in `compute_out_given_in` function. But we also have good tests for `constant_product.rs` so probably not a big deal.

Reply:
Are you concerned that we don't have substantial testing to ensure that weights are being respected by the dex math library?